File: garage-app/src/app/interface/Car.ts
import { CarDetails } from './CarDetails';

export class Car {
    id: number;
    year: number;
    model: string;
    make: string;
    price: string;
    licensed: boolean;
    dateAdded: string;
    carDetails: CarDetails;

    constructor(id: number, year: number, model: string, make: string, price: string, licensed: boolean, dateAdded: string, carDetails: CarDetails) {
        this.id = id;
        this.year = year;
        this.model = model;
        this.make = make;
        this.price = price;
        this.licensed = licensed;
        this.dateAdded = dateAdded;
        this.carDetails = carDetails
    }
}

File: garage-api/src/main/java/com/garage/entity/Car.java
package com.garage.entity;

import java.math.BigDecimal;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Car {
	
	@Id
	@Min(1)
	@NotNull
	private Integer id;
	
	@Min(1500)
	@NotNull
	private int year;
	
	@NotBlank(message = "Model is mandatory")
	private String model;
	
	@NotBlank(message = "Make is mandatory")
	private String make;
	
	@Min(1)
	@NotNull
	private BigDecimal price;
}
